### PR TITLE
Remove enable_brave_sync buildflag

### PR DESCRIPTION
**Pull request by darkdh**
Resolves https://github.com/brave/brave-browser/issues/17533

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/BraveSyncReflectionUtils.java [hunk 1]
```diff
@@ -21,26 +21,19 @@
...
-                try {
-                    sBraveSyncWorker =
-                        Class.forName("org.chromium.chrome.browser.BraveSyncWorker")
-                             .getConstructor()
-                             .newInstance();
-                } catch (Exception e) {
-                    Log.e(TAG, "Cannot create BraveSyncWorker ", e);
-                }
+            try {
+                sBraveSyncWorker = Class.forName("org.chromium.chrome.browser.BraveSyncWorker")
```
Comment (AlexeyBarabash) [inline android/java/org/chromium/chrome/browser/BraveSyncReflectionUtils.java:25]: This whole file should be removed. If we now don't have `BraveConfig.SYNC_ENABLED` - we can always create `new BraveSyncWorker()` and invoke `BraveSyncInformers.show()` and don't need to create classes with reflection.

Places with call `BraveSyncReflectionUtils.showInformers()` should be changed to `BraveSyncInformers.show()`.

And `BraveSyncReflectionUtils.getSyncWorker()` should be moved to static `BraveSyncWorker.get()` and be replaced over the code.

**Review by AlexeyBarabash — APPROVED** [review]
Commit 1180c1740659b5f5a679c80bcbee3119a0756e01 looks good.

But I can't review my own commit.

File: browser/android/BUILD.gn [hunk 2]
```diff
@@ -5,23 +5,35 @@ source_set("android_browser_process") {
   check_includes = false
 
   sources = [
+    "//brave/browser/sync/brave_sync_devices_android.cc",
```
Comment (bridiver) [inline browser/android/BUILD.gn:8]: Do not include sources from other directories like this. Either move this file or create a dep and add the dep instead

Comment (bridiver) [reply browser/android/BUILD.gn:8]: I guess this is copied, but can we get a follow-up for it? We really shouldn't have things like this

**Review by bridiver — APPROVED** [review]

File: browser/android/BUILD.gn (hunk 2 above)
Comment (darkdh) [reply browser/android/BUILD.gn]: fixed in 618ac477ab3f2ecaef41b1849d8ed22b7113ab57

**Pull request merged**